src/main/java/io/reactivex/internal/operators/completable/CompletableFromSource.java

```diff
@@ -13,18 +13,62 @@
 
 package io.reactivex.internal.operators.completable;
 
-import io.reactivex.*;
-
-public final class CompletableFromSource extends Completable {
-
-    final CompletableSource source;
+import io.reactivex.CompletableObserver;
+import io.reactivex.CompletableSource;
+import io.reactivex.disposables.Disposable;
+import java.util.concurrent.atomic.AtomicBoolean;
 
+public final class CompletableFromSource extends CompletableFromUnsafeSource {
     public CompletableFromSource(CompletableSource source) {
-        this.source = source;
+        super(source);
+    }
+
+    @Override protected void subscribeActual(CompletableObserver observer) {
+        super.subscribeActual(new DisposeAwareCompletableObserver(observer));
```

Comment:
It prevents notifications after disposal for implementations which might not be checking.